server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.Strings;
+
+public class MapperBuilderContext {
+
+    public static MapperBuilderContext root() {
+        return new MapperBuilderContext(null);
+    }
+
+    // TODO remove this
+    public static MapperBuilderContext forPath(ContentPath path) {
```

Comment:
Can this already be removed? It mostly seems to be about removing a potential dot at the end. I see uses in DynamicFieldsBuilder and FieldMapper#merge, and at least the later seems to avoind having a dot at the end through the way ContentPath#parentPath works, so the helper could go to DynamicFieldsBuilder if it is really still needed there.

Reply:
It's not just about the dot, it's about creating a new context for an arbitrary path, which we currently need due to the way merging works.  Once I have moved merging entirely to Builders then it can go.